src/app/modals/encryptwallet/encryptwallet.component.ts

```diff
@@ -41,20 +41,19 @@ export class EncryptwalletComponent {
         this._rpc.call('encryptwallet', [password.password])
           .subscribe(
             response => {
-              this._rpc.state.set('ui:spinner', false);
               this._rpc.toggleState(false);
               this.flashNotification.open(response);
 
               if (this._rpc.isElectron) {
                 this._rpc.call('restart-daemon')
                   .subscribe(() => {
+                    this._rpc.state.set('ui:spinner', false);
                     if (!this._modalsService.initializedWallet) {
                       this._modalsService.open('createWallet', {forceOpen: true});
-                    } else {
-                      this._modalsService.close();
-                      // force-close encrypt modal
-                      this._dialogRef.close();
                     }
+                    // force-close encrypt modal
+                    this._dialogRef.close();
```

Comment:
This looks a bit odd..

Reply:
Both dialogRef & modelService are being used to close the dialog.

Reply:
I just deleted else condition, yes no need to two close functions. but we need to keep dialog ref close because that time modal service is not able to close encryption modal !!